### DISCUSSION TITLE
Help with Rec601YCbCr to sRGB conversion

### DISCUSSION THREAD

**Discussion by Kargo** [dlemstra/Magick.NET#1171] [Help] [answered]
I'm trying to convert a JP2 image in the Rec601YCbCr colorspace to an sRGB JPEG, while also adding some white borders.

Running the following command in the command line produces the result that I want.
`magick file3.jp2 -colorspace sRGB -extent 1000x1000 output.jpg`

How would I replicate this in code? So far I have been unable to reproduce the extent effect without the background of the output image being pink.

**Comment by dlemstra** [comment]
What have you tried and did not work?

**Comment by Kargo** [comment, marked as answer]
This is the result without touching the colorspace
![image](https://user-images.githubusercontent.com/5476181/161205350-ec1a3ffd-7b46-46ca-af6e-569217091a66.png)

I've done previous colour space conversions (like sRGB to CMYK) by using the SetProfile method.
```csharp
image.SetProfile(ColorProfile.SRGB);
image.SetProfile(ColorProfile.USWebCoatedSWOP);
```

There is no built in profile for Rec601YCbCr, so I've downloaded an Rec601 ICC profile [here](https://github.com/saucecontrol/Compact-ICC-Profiles/tree/master/profiles).

```csharp
image.SetProfile(new ColorProfile(@"\icc\Rec601PAL-v4.icc"));
image.SetProfile(ColorProfile.SRGB);
```
This gives me the white background that I want, but it also ruins the image's colours.
![image](https://user-images.githubusercontent.com/5476181/161205193-c33ffdee-71a8-48a3-bcf6-99a3d5c079b4.png)

I can send the problematic image privately, but I'm unfortunately not allowed to post it publicly.

**Reply by dlemstra** [reply]
The `-colorspace sRGB` option is the same as `image.ColorSpace = ColorSpace.sRGB`. Have you tried using that instead? Another option would be to use `image.TransformColorSpace` together with a profile.

**Reply by Kargo** [reply]
Thank you, after some more tinkering I got it working by using `image.ColorSpace = ColorSpace.sRGB`.
